ldap_auth_provider.py

```diff
@@ -124,7 +124,7 @@ def check_password(self, user_id, password):
                 )
                 if not result:
                     defer.returnValue(False)
-            else:
+            else:  # pragma: no cover
```

Comment:
I'm a bit dubious about whether some of these `no cover`s are justified, but don't have a hugely strong opinion.

Reply:
I was mostly taking the view that since we ensure that this can't happen when we create the ldap provider its basically impossible to test for, so not much point adding it to the coverage? But shrug.